backend/routes/knowledge.py

```
@@ -0,0 +1,745 @@
+        if not hasattr(retriever, "semantic_searcher") or not retriever.semantic_searcher:
+            raise HTTPException(status_code=503, detail="Semantic searcher not available")
+
+        if not retriever.semantic_searcher.vector_store:
+            raise HTTPException(status_code=503, detail="Vector store not available")
+
+        # Get the collection from the vector store
+        try:
+            collection = retriever.semantic_searcher.vector_store._collection
+        except Exception as e:
+            logger.error(f"Failed to get collection: {e}")
+            raise HTTPException(status_code=503, detail="Collection not available")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)